app/controllers/adopter_profiles_controller.rb

```diff
@@ -46,6 +46,12 @@ def update
     end
   end
 
+  def states
+    @target = params[:target]
+    @name = params[:name]
+    @states = CS.states(params[:country]).invert
```

Comment:
I think we can make a seperate controller called
CountriesController in which you can make this route:

```
GET /countries/:country/states
```
This is a bit more restful and communicates a bit more clearly what resource you'd expect. In this case, "states"